WordPress/Classes/System/3DTouch/WP3DTouchShortcutCreator.swift

```diff
@@ -26,15 +26,23 @@ public class WP3DTouchShortcutCreator: NSObject
         blogService = BlogService(managedObjectContext: mainContext)
     }
     
-    public func createShortcuts(loggedIn: Bool) {
+    public func createShortcutsIf3DTouchAvailable(loggedIn: Bool) {
+        if !is3DTouchAvailable() {
+            return
+        }
+        
         if loggedIn {
-            createLoggedInShortcuts()
+            if hasBlog() {
+                createLoggedInShortcuts()
+            } else {
+                clearShortcuts()
+            }
```

Comment:
I think this is problematic for two reasons:
1. Even if you have no blogs, it still makes sense to show the notifications shortcut
2. In the `hasBlog` case, you still have to deal with the possibility of `lastUsedOrFirstBlog` being `nil` at compile time, even if you know it can never happen.

I'd revert this bit to the previous version and change `createLoggedInShortcuts` instead:

```swift
private func loggedInShortcutArray() -> [UIApplicationShortcutItem] {
    let notificationsShortcut = ...
    if let defaultBlog = blogService.lastUsedOrFirstBlog() {
        let defaultBlogName = defaultBlog.settings?.name
        // Create stats, newPhoto, and newPost shortcuts
        // Return [notifications, stats, newPhoto, newPost]
    } else {
        // Return [notificationsShortcut]
    }
```

I've split the `defaultBlog` optional check in two parts because you can have one blog with a `nil` name. In that case you'd want to show all the shortcuts even if you don't show the site name.